### DISCUSSION TITLE
Select component slow

### DISCUSSION THREAD

**Discussion by yohannes81** [wireui/wireui#182] [Q&A]
Hi
when I use select component with for example 1000 options value,
the page is completely slow .. everything freezes. is this possible?
how can I do? I need the select with the search
thanks for u help

<x-select
    label="Select Status"
    placeholder="Select one status"
    wire:model.defer="model"
>
    <x-select.option label="Pending" value="1" />
    <x-select.option label="In Progress" value="2" />
    <x-select.option label="Stuck" value="3" />
    <x-select.option label="Done" value="4" />
    ...
   <x-select.option label="Done" value="999" />
   <x-select.option label="Done" value="1000" />
</x-select>

**Comment by REJack** [comment]
Did you tested it on a production server or on your (local) development server?

**Reply by yohannes81** [reply]
local

**Reply by REJack** [reply]
I would recommend to test it on a production server, I have often slow performance on my local dev server.

**Reply by REJack** [reply]
Also disable DevTools that causes so much CPU consumtion while Livewire developing 😄

**Reply by yohannes81** [reply]
without the DevTools much better 
when i finish i will see in production 
thanks

**Reply by REJack** [reply]
No problem 😄 Modals with DevTools are horrible too 🤣

**Reply by ph7jack** [reply]
@yohannes81 the problem still persists?

**Reply by bbcreation** [reply]
I have the same problem. 3k records. Page takes 6sec. Air Mac M1. Can we count on the async option on version 1?

**Comment by ph7jack** [comment]
Guys, I have released the async search in the select component, just update your project, read the documentation and implement your API.
Using the async-search, you can't use the slot options, you need to use the option-label and the option-value attributes
https://livewire-wireui.com/docs/select
